packages/enclave-contracts/contracts/token/EnclaveToken.sol

```diff
@@ -0,0 +1,233 @@
+    function mintAllocation(
+        address recipient,
+        uint256 amount,
+        string memory allocation
+    ) external onlyRole(MINTER_ROLE) {
+        if (recipient == address(0)) revert ZeroAddress();
+        if (amount == 0) revert ZeroAmount();
+        // Ensure we do not exceed the total supply.
+        if (totalMinted + amount > TOTAL_SUPPLY) revert ExceedsTotalSupply();
+
+        _mint(recipient, amount);
+        totalMinted += amount;
+        emit AllocationMinted(recipient, amount, allocation);
+    }
+
+    /**
+     * @notice Mint multiple allocations in a batch.
+     * @dev Only accounts with the MINTER_ROLE may call this function.
+     * @param recipients Array of addresses to receive tokens.
+     * @param amounts Corresponding array of amounts to mint.
+     * @param allocations Array of allocation descriptions.
+     */
+    function batchMintAllocations(
+        address[] memory recipients,
+        uint256[] memory amounts,
+        string[] memory allocations
+    ) external onlyRole(MINTER_ROLE) {
+        if (
+            recipients.length != amounts.length ||
+            amounts.length != allocations.length
+        ) revert ArrayLengthMismatch();
+
+        uint256 totalAmount = 0;
+        for (uint256 i = 0; i < amounts.length; i++) {
+            totalAmount += amounts[i];
+        }
+        if (totalMinted + totalAmount > TOTAL_SUPPLY)
+            revert ExceedsTotalSupply();
+
+        for (uint256 i = 0; i < recipients.length; i++) {
+            address rec = recipients[i];
+            uint256 amt = amounts[i];
+            if (rec == address(0)) revert ZeroAddress();
+            if (amt == 0) revert ZeroAmount();
+
+            _mint(rec, amt);
+            emit AllocationMinted(rec, amt, allocations[i]);
+        }
+        totalMinted += totalAmount;
+    }
+
+    /**
+     * @notice Enable or disable transfer restrictions.
+     * @dev Only the owner can toggle this flag.
+     * @param restricted Whether transfers should be restricted.
+     */
+    function setTransferRestriction(bool restricted) external onlyOwner {
+        transfersRestricted = restricted;
+        emit TransferRestrictionUpdated(restricted);
+    }
+
+    /**
+     * @notice Add or remove an address from the transfer whitelist.
+     * @dev Only the owner may call this.
+     * @param account Address whose whitelist status is to be updated.
+     * @param whitelisted Whether the address should be whitelisted.
+     */
+    function setTransferWhitelist(
+        address account,
+        bool whitelisted
+    ) external onlyOwner {
+        transferWhitelisted[account] = whitelisted;
+        emit TransferWhitelistUpdated(account, whitelisted);
+    }
+
+    /**
+     * @notice Toggle an account's whitelist status.
+     * @dev Only the owner may call this.
+     * @param account Address whose whitelist status should be toggled.
+     */
+    function toggleTransferWhitelist(address account) external onlyOwner {
```

Comment:
do we need setTransferWhitelist if we have this toggle function that can do both?